tests/unit/backend/corpora/api_server/test_v1_dataset.py

```diff
@@ -112,6 +112,14 @@ def test__minimal_status__ok(self):
             response = self.app.get(test_url.url, headers=headers)
             self.assertEqual(json.loads(response.data)["upload_status"], status.name)
 
+    def test__get_all_datasets_for_index(self):
+        test_url = furl(path="/dp/v1/datasets/index")
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.get(test_url.url, headers=headers)
+        self.assertEqual(200, response.status_code)
+        # Note: testing the payload is difficult because this will retrieve datasets generated by previous
+        #  tests.
```

Comment:
could _add_ new dataset and then at least assert that it is included and and correctly represented.  could also compare the length of the json array to the actual db table row count.

Reply:
The latter would be a good idea but I don't think we have access to the raw database session here. Ideally we would make the test independent although that will require some research (moving it to a separate class doesn't help). I think the idea of adding a dataset and assert that the id is there, as well as the relevant fields, is a good idea.

Reply:
You can reset the state of the DB using this https://github.com/chanzuckerberg/single-cell-data-portal/blob/18021ed87f7365b6061b3036f34ea4fab12a87e6/tests/unit/backend/fixtures/data_portal_test_case.py#L23

Reply:
I tried it but I couldn't get it to work. I refactored the test so that it will work regardless, but I'd like to come back to this at some point.